python/ray/experimental/workflow/tests/test_fork_join.py

```diff
@@ -0,0 +1,39 @@
+from ray.experimental import workflow
+from ray.experimental.workflow.tests.utils import resolve_workflow_output
+
+
+@workflow.step
+def bar0():
+    return '[bar0]'
+
+
+@workflow.step
+def bar1(x):
+    return x + '[bar1]'
+
+
+@workflow.step
+def bar2(x):
+    return x + '[bar2]'
+
+
+@workflow.step
+def simple_join_task(x, y):
+    return f"bar3({x}, {y})"
+
+
+@workflow.step
+def fork_join():
+    x = bar0.step()
+    x = bar1.step(x)
+    y = bar1.step('#')
+    return simple_join_task.step(x, y)
+
+
+def test_fork_join():
+    import ray
+    ray.init()
+
+    outputs = workflow.run(fork_join.step())
+    assert resolve_workflow_output(outputs) == "bar3([bar0][bar1], #[bar1])"
```

Comment:
```suggestion
    assert ray.get(outputs) == "bar3([bar0][bar1], #[bar1])"
```

Can we change the tests to use `ray.get()` instead? Any resolution should be done internally by workflow.run().

Reply:
I think the current design is to return ObjectRef, this prevents storing results in the driver script and make use more capable to chain the result to another workflow. @iycheng how do you think of it?

Reply:
Sure. I imagine this will have to change later. Or, the resolution can be done by inserting post-processing code for the output of the function.